lib/utility/recipe_puppy_parser.rb

```diff
@@ -0,0 +1,91 @@
+module Utility
+  class RecipePuppyParser
+    ALL_RECIPES_URL = 'http://www.recipepuppy.com/api/?q=&p='.freeze
+    BASE_URL = 'http://www.recipepuppy.com/api/'.freeze
+
+# API query, not a database query
+    def query_all_recipes_for(*ingredients)
+      begin
+        ing_search = ingredients.join('').gsub(/\s+/, "")
+        recipes = []
+        100.times do |page|
+          response = HTTParty.get("#{BASE_URL}?i=#{ing_search}&p=#{page + 1}")
+          if response.success?
+            parsed = JSON.parse(response)
+            parsed["results"].each do |recipe|
+              searched_recipe = Recipe.find_by(name: recipe["title"])
+              if searched_recipe
+                recipes << searched_recipe
+              end
+            end
+          end
+        end
+        recipes.each do |r|
+          puts "Recipe: #{r.name}"
```

Comment:
remember that function like this are not really needed anymore. If our method can return an array of recipes, then our frontend can accept that array and use it. No puts-ing required here